### ISSUE TITLE
`pl$DataFrame()` should accept a `schema` argument

### ISSUE THREAD

**Issue by etiennebacher** [eitsupi/neo-r-polars#13]
```r
pl$DataFrame(a = 1:3, schema = list(a = pl$Int16))
#> Error in `as_polars_df()` at neo-r-polars/R/dataframe-frame.R:55:3:
#> ! Evaluation failed.
#> Caused by error in `FUN()`:
#> ! Unsupported class for `as_polars_series()`: polars_data_type, polars_object
```

**Comment by eitsupi** [comment]
To simplify the implementation, could you call the cast method later?
Since unlike Python lists, R vectors are guaranteed to have all elements of the same type and the type of the corresponding Polars is fixed, which seems more natural.

**Comment by etiennebacher** [comment]
It would easier to translate code from py-polars to r-polars if we accepted this argument (it would also save some time by not having to move code to `$cast()`

**Comment by eitsupi** [comment]
If we were to add an argument named `schema` to `pl$DataFrame()`, we would not be able to create a column named `schema`, so in the tidyverse world it is common to use an argument name like `.schema`.
Is that reasonable?

**Comment by etiennebacher** [comment]
Yes I think it would be ok

**Comment by eitsupi** [comment]
Hmmm, I looked at it a bit and it looks more complicated than I thought to implement.
I'm busy this week so I'll look at it next week.

**Comment by eitsupi** [comment]
Thinking about it again, isn't there a difference between `schema` and `schema_override` here, and what you really want is `schema_override`, not `schema`, I think.
See pola-rs/r-polars#897

**Comment by eitsupi** [comment]
Closed by 29ff4d2183b301985b23880e3ea3ab607533c489
